rust/telemetry/src/lib.rs

```diff
@@ -1,5 +1,6 @@
 use std::{sync::Arc, time::Duration};
 
+use env::SELF_HOSTED;
```

Comment:
```suggestion
use env::ON_PREM;
```

One small change for semantics, one giant leap for perceived value 🧑‍🚀

Reply:
(I know it's internal, but probably better to align on this naming strategy going forward)